### ISSUE TITLE
HTML completion

### ISSUE THREAD

**Issue by wizardlink** [w0rp/ale#2518]
Hello, I am almost certain that this is already a feature. However I could not find an answer yet to this. As I am short on time, I figured that opening an issue could be the best option.

Is there a way to have the same completion capabilities that ALE gives in TS and JS on HTML? In this matter, whenever I type properties like `class`, `src` or even tags... ALE does not gives any suggestions. It is not a big deal but it would certainly shorten my work if I could do it without `ins-completion`.

Thanks in advance!

**Comment by w0rp** [comment]
ALE doesn't do the work to provide completion data itself, it connects to language servers. This could be supported by adding support for the VSCode HTML language server: https://github.com/vscode-langservers/vscode-html-languageserver-bin I'll update this issue to note that.

**Comment by wizardlink** [comment]
I see, I am sort of new to VIM (more specifically, I am still mining the beginning of the user manual) and specially ALE. Just to clarify, other languages when connecting to the "proper" language server will provide that keyword completion while typing, correct?

Sorry for making this question, I am actually very intrigued and want to learn as most as possible. Thanks in advance!

**Comment by w0rp** [comment]
Yeah, a language server could be used for providing completion data. You can also look for completion plugins like Deoplete which pull in completion data from other sources too.

**Comment by wizardlink** [comment]
I see, thanks for the clarification! Would you recommend me Deoplete over ALE's completion for TS as well? I know you work daily with TS so I am sure that you know the ups and downs of it better than I do.

**Comment by w0rp** [comment]
Deoplete is a plugin which collects completion data from various sources, and there is a Deoplete completion source in ALE. There are also some other completion plugins like mucomplete and you can use them to combine completion data with ALE's `omnifunc` function. (`:help ale#completion#OmniFunc()`)

**Comment by wizardlink** [comment]
That is awesome, thanks. Do not get me wrong though, ALE is the last plugin I would ditch. I appreciate all the effort you put into it! o/
